src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -23,52 +24,172 @@
 
 (def EnvironmentClassesFileEntry
   "Schema for an individual file entry which is part of the return payload
-  for an environment_classes request"
+  for an environment_classes request."
   {:path schema/Str
-   :classes [Map]})
+   schema/Keyword schema/Any})
 
 (def EnvironmentClassesInfo
-  "Schema for the return payload an environment_classes request"
+  "Schema for the return payload an environment_classes request."
   {:name schema/Str
    :files [EnvironmentClassesFileEntry]})
 
+(defn sort-nested-java-maps
+  "For a data structure, recursively sort any nested maps and sets descending
+  into map values, lists, vectors and set members as well. The result should be
+  that all maps in the data structure become explicitly sorted with natural
+  ordering. This can be used before serialization to ensure predictable
+  serialization.
+
+  The returned data structure is not a transient so it is still able to be
+  modified, therefore caution should be taken to avoid modification else the
+  data will lose its sorted status.
+
+  This function was copypasta'd from clj-kitchensink's core/sort-nested-maps.
+  sort-nested-maps can only deep sort a structure that contains native Clojure
+  types, whereas the data structure that we're sorting in this case contains
+  objects created by JRuby that derive from standard Java interfaces.  This
+  function tolerates Java-derived objects when considering how to apply
+  sorting."
+  [data]
+  (cond
+    (instance? Map data)
+    ;; sorted-map expects each key to be a 'keyword'.  Some of the keys
+    ;; that we get back from JRuby might not be of a type that Clojure's
+    ;; 'keyword' function can turn directly into a Clojure keyword, e.g., a
+    ;; RubySymbol, so .toString is used to get a String representation of
+    ;; the key object, which can then be turned into a Clojure keyword.
+    (into (sorted-map) (for [[k v] data]
+                         [(or (keyword k)
+                              (keyword (.toString k)))
+                          (sort-nested-java-maps v)]))
+
+    (instance? Iterable data)
+    (map sort-nested-java-maps data)
+
+    :else data))
+
+(schema/defn ^:always-validate
+  if-none-match-from-request :- (schema/maybe String)
+  "Retrieve the value of an 'If-None-Match' HTTP header from the supplied Ring
+  request.  If the header is not found, returns nil."
+  [request :- {schema/Keyword schema/Any}]
+  (rr/get-header request "If-None-Match"))
+
+(schema/defn ^:always-validate
+  basic-manifest-info-from-jruby->basic-manifest-info-for-json
+    :- {schema/Any schema/Any}
+  "Convert the value for a manifest file entry into an appropriate map for
+  use in serializing an environment_classes response to JSON."
+  [file-info :- Map$Entry]
+  (let [file-detail (val file-info)]
+    (if (instance? Map file-detail)
```

Comment:
I expect this logic to go away soon - when [SERVER-1126](https://tickets.puppetlabs.com/browse/SERVER-1126) is done.  After that point, we should just be able to coerce the original RubyHash to a Clojure map rather than conditionally wrapping the content in a Map with a `:classes` key.